### ISSUE TITLE
Pagination issue when filtering for multiple labels

### ISSUE THREAD

**Issue by subesokun** [jdalrymple/gitbeaker#2322]
**Description**

`Issues.all()` seems to have an issue with pagination when filtering for multiple labels. Example:

```Javascript
await api.Issues.all({
    projectId: 123,
    labels: ["Issue::Incident", "Vulnerability::*"],
    state: "opened"
  })
```

When setting multiple labels, it fetches always only the first page (by default 20 issues) and stops that after. In my case there are 190 issues that meet the search criteria. To verify that this not a bug in the GitLab API, I've run a curl request against the API and the pagination headers are correct in that case.

```
curl --head --header "PRIVATE-TOKEN: ${mytoken}" "https://mygitlab.com/api/v4/projects/123/issues?labels=Issue::Incident,Vulnerability::*&state=opened"

x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-next-page: 2
x-page: 1
x-per-page: 20
x-prev-page:
x-total: 190
x-total-pages: 10
```

**Steps to reproduce**

Create > 20 issues and assign them the labels `a` and `b`

```Javascript
const result = await api.Issues.all({
    projectId: 123,
    labels: ["a", "b"],
    state: "opened"
  })
console.log(result.length)
```

Will always print `20`

**Expected behaviour**

`all()` fetches all results.

**Actual behaviour**

`all()` fetches only the first page.

**Possible fixes**

None?

**Comment by jdalrymple** [comment]
Ill give it a look! I should be looking at the Labels API soon anyways in #2258

**Comment by jdalrymple** [comment]
Found the error. Its not a gitbeaker issue perse. Gitlab doesnt return the link header "next" and thus, the library doesnt know how to paginate correctly. Depending on the pagination headers you mentioned above is generally bad practice. I will fire off a bug report

**Comment by jdalrymple** [comment]
I just tested this locally with a test repo and it worked with the usecase you mentioned.

**Comment by subesokun** [comment]
Just updated to the latest version but still having the issue. How many issues did you generate? Our system currently reports ~190 issues. All of them have the label `a` and `b` (to be exact once scoped label `a::Open` and a simple label `b`). If I query for `a` or `b`  separately, I get all issues. If I query for `a` & `b`, I receive only 20 :(

```typescript
async function listUnresolvedIssues(gitlabApi: any, issueProjectId: number) {
  return gitlabApi.Issues.all({
    projectId: issueProjectId,
    labels: ["A::Open", "B"],
    state: "opened"
  })
}
```

Is there a way to generate some kind of debug messages that might be helpful for you?

**Comment by subesokun** [comment]
Made a small demo that reproduces the issue

```javascript
const { Gitlab } = require('@gitbeaker/node');

async function main() {
    const gitlabApi = new Gitlab({
        host: 'https://gitlab.com',
        token: 'yourPAT',
    })

    // https://gitlab.com/subesokun/gitbeaker-issue-2322
    const issues = await gitlabApi.Issues.all({
        projectId: '33484309',
        labels: ['A::Open', 'B'],
        state: 'opened'
    })

    // prints 20 instead of 21
    console.log(issues.length)
}

main()
```

**Comment by jdalrymple** [comment]
It may of been due to a older version i was running locally :facepalm:  Investigating now! Thanks again for the additional info

**Comment by jdalrymple** [comment]
It does look like a weird query argument formatting. Fixing it now

**Comment by jdalrymple** [comment]
It also looks like not all query parameters follow the same array format. Some are indexed, some are commas which is frustrating.

**Comment by subesokun** [comment]
Awesome, thanks! Looking forward to the next release :)

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`35.4.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/35.4.0) :rocket:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`37.0.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/37.0.0) :rocket:
